File: app/src/main/java/com/nordokod/scio/constants/Exceptions.java
package com.nordokod.scio.constants;

public enum Exceptions {
    INVALID_VALUE_EXCEPTION("SDE-01","Valor Inválido");
    private String Code;
    private String Title;
    Exceptions(String code,String Title){
        this.Code=code;
        this.Title=Title;
    }
    public String getCode(){
        return this.Code;
    }
    public String getTitle(){
        return this.Title;
    }
}

File: app/src/main/java/com/nordokod/scio/kt/constants/Exceptions.kt
package com.nordokod.scio.kt.constants

class UnknownException: Exception()

class IrrelevantException: Exception()

class PhoneNetworkException: Exception()

class OnlyPremiumException: Exception()

class GuideException(val code: Code): Exception(){
    enum class Code{
        NO_GUIDES, NO_QUESTIONS_IN_GUIDE,GUIDE_NOT_AVAILABLE
    }
}

class InputDataException(val code: Code): Exception(){
    enum class Code {
        DATETIME_AFTER, DATETIME_BEFORE, INVALID_MAIL, INVALID_PASSWORD, INVALID_USERNAME, PASSWORDS_NOT_MATCH, EMPTY_FIELD, NOT_ENOUGH_OPTIONS, NOT_CORRECT_OPTION_SELECTED
    }
}